runtime/src/accounts_db.rs

```diff
@@ -3310,15 +3310,18 @@ impl AccountsDb {
         slot: Slot,
         hashes: &[impl Borrow<Hash>],
         mut storage_finder: F,
-        accounts_and_meta_to_store: &[(StoredMeta, &AccountSharedData)],
+        accounts_and_meta_to_store: &[(StoredMeta, Option<&AccountSharedData>)],
```

Comment:
What do you think about an enum `CommitAccountSharedData` that has two variants, `Dirty` and `Unmodified`?

Reply:
I think the 2 options here mean:
1. zero lamport account intended only to hide earlier zero lamport accounts (None)
2. every other account we want to write to storage or cache